src/source/Sdp/Serialize.c

```diff
@@ -1,160 +1,182 @@
-#define LOG_CLASS "SDP"
+#define LOG_CLASS "SDPSerialize"
 #include "../Include_i.h"
+#include "kvssdp/sdp_serializer.h"
 
-STATUS serializeVersion(UINT64 version, PCHAR* ppOutputData, PUINT32 pTotalWritten, PUINT32 pBufferSize)
+// Convert error code from SDP library to STATUS.
+extern STATUS convertSdpErrorCode(SdpResult_t sdpResult);
+
+static STATUS serializeVersion(SdpSerializerContext_t* pCtx, UINT64 version)
 {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
-    UINT32 currentWriteSize = 0;
-
-    currentWriteSize = SNPRINTF(*ppOutputData, (*ppOutputData) == NULL ? 0 : *pBufferSize - *pTotalWritten,
-                                SDP_VERSION_MARKER "%" PRIu64 SDP_LINE_SEPARATOR, version);
+    SdpResult_t sdpResult = SDP_RESULT_OK;
 
-    CHK(*ppOutputData == NULL || ((*pBufferSize - *pTotalWritten) >= currentWriteSize), STATUS_BUFFER_TOO_SMALL);
-    *pTotalWritten += currentWriteSize;
-    if (*ppOutputData != NULL) {
-        *ppOutputData += currentWriteSize;
-    }
+    sdpResult = SdpSerializer_AddU64(pCtx, SDP_TYPE_VERSION, version);
 
-CleanUp:
+    retStatus = convertSdpErrorCode(sdpResult);
 
     LEAVES();
     return retStatus;
 }
 
-STATUS serializeOrigin(PSdpOrigin pSDPOrigin, PCHAR* ppOutputData, PUINT32 pTotalWritten, PUINT32 pBufferSize)
+static STATUS serializeOrigin(SdpSerializerContext_t* pCtx, PSdpOrigin pSDPOrigin)
```

Comment:
For what reason are we making these functions static?

Reply:
These functions are only used inside Serialize.c. It's more secure to keep it in file scope with static keyword IMO. Or does this function open for external use?

Reply:
I suppose previously nothing blocked it from being available for external use.  I like the idea of making it static.